File: laravel_api/app/Models/Language.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class Language extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = ['name','code'];

    public function translation(){
        return $this->hasMany(Translation::class);
    }
    public function menu()
    {
        return $this->belongsToMany(Menu::class);
    }
    public function users(){
        return $this->belongsToMany(User::class,'language_user');
    }
    public function user_config(){
        return $this->hasMany(LanguageUser::class);
    }
}

File: menu_flutter/lib/models/app/language.dart
class Language {
  Language({
    required this.id,
    required this.name,
    required this.code,
  });

  int id;
  String name;
  String code;

  static Language fromJson(dynamic obj) {
    return Language(
      id: obj['id'],
      name: obj['name'],
      code: obj['code'],
    );
  }
}
